code/datums/status_effects/status_effect.dm

```diff
@@ -0,0 +1,120 @@
+
+//Status effects are used to apply temporary or permanent effects to mobs. Mobs are aware of their status effects at all times.
+//This file contains their code, plus code for applying and removing them.
+//When making a new status effect, add a define to status_effects.dm in __DEFINES for ease of use!
+
+/datum/status_effect
+	var/id = "effect" //Used for screen alerts.
+	var/duration = -1 //How long the status effect lasts in DECISECONDS. Enter -1 for an effect that never ends unless removed through some means.
+	var/tick_interval = 10 //How many deciseconds between ticks, approximately. Leave at 10 for every second.
+	var/mob/living/owner //The mob affected by the status effect.
+	var/status_type = STATUS_EFFECT_UNIQUE //How many of the effect can be on one mob, and what happens when you try to add another
+	var/on_remove_on_mob_delete = FALSE //if we call on_remove() when the mob is deleted
+	var/examine_text //If defined, this text will appear when the mob is examined - to use he, she etc. use "SUBJECTPRONOUN" and replace it in the examines themselves
+	var/alert_type = /obj/screen/alert/status_effect //the alert thrown by the status effect, contains name and description
+	var/obj/screen/alert/status_effect/linked_alert = null //the alert itself, if it exists
+
+/datum/status_effect/New(list/arguments)
+	on_creation(arglist(arguments))
+
+/datum/status_effect/proc/on_creation(mob/living/new_owner, ...)
+	if(new_owner)
+		owner = new_owner
+	if(owner)
+		LAZYADD(owner.status_effects, src)
+	if(!owner || !on_apply())
+		qdel(src)
+		return
+	if(duration != -1)
+		duration = world.time + duration
+	tick_interval = world.time + tick_interval
+	if(alert_type)
+		var/obj/screen/alert/status_effect/A = owner.throw_alert(id, alert_type)
+		A.attached_effect = src //so the alert can reference us, if it needs to
+		linked_alert = A //so we can reference the alert, if we need to
+	processing_objects.Add(src)
+	fast_processing.Add(src)
```

Comment:
I'm not sure it's safe or a good idea to put this on both the processing_objects list and fast_processing